kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -127,6 +132,158 @@ matchOverloading termPair =
             Simple pair -> return pair
             _ -> notApplicable
 
+-- flipPairBack :: Pair a -> Pair a
+-- flipPairBack (Pair x y) = Pair y x
+
+data UnifyOverload1 = UnifyOverload1
+    { firstHead, secondHead :: Symbol
+    , firstChildren :: [TermLike RewritingVariableName]
+    , inj :: Inj (TermLike RewritingVariableName)
+    }
+
+data UnifyOverload2 = UnifyOverload2
+    { firstHead, secondHead :: Symbol
+    , secondChildren :: [TermLike RewritingVariableName]
+    , inj :: Inj (TermLike RewritingVariableName)
+    }
+
+data UnifyOverload3 = UnifyOverload3
+    { firstHead, secondHead :: Symbol
+    , firstChildren, secondChildren :: [TermLike RewritingVariableName]
+    , inj :: Inj (TermLike RewritingVariableName)
+    }
+
+data UnifyOverload4 = UnifyOverload4
+    { firstHead :: Symbol
+    , secondVar :: ElementVariable RewritingVariableName
+    , inj :: Inj (TermLike RewritingVariableName)
+    }
+
+data UnifyOverload5 = UnifyOverload5
+    { firstHead :: Symbol
+    , firstChildren :: [TermLike RewritingVariableName]
+    , secondVar :: ElementVariable RewritingVariableName
+    , inj :: Inj (TermLike RewritingVariableName)
+    }
+
+data UnifyOverload6 = UnifyOverload6
+    { firstHead :: Symbol
+    , injChild :: TermLike RewritingVariableName
+    }
+
+data OverloadedConstructorSortInjectionAndEquals
+    = Overload1 UnifyOverload1
+    | Overload2 UnifyOverload2
+    | Overload3 UnifyOverload3
+    | Overload4 UnifyOverload4
+    | Overload5 UnifyOverload5
+    | Overload6 UnifyOverload6
```

Comment:
These all need real names.

Reply:
This code isn't currently being used in this PR, should I just remove it for now?